src/Microsoft.Health.Dicom.Functions.UnitTests/Management/PurgeOrchestrationInstancesHistoryTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Functions.Management;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Functions.UnitTests
+{
+    public class PurgeOrchestrationInstancesHistoryTests
+    {
+        private readonly PurgeOrchestrationInstancesHistoryConfiguration _purgeConfig;
+        private readonly TimerInfo _timer;
+        private readonly ILogger _logger;
+        private readonly DateTime _definedNow;
+        private readonly PurgeOrchestrationInstancesHistory _purgeOrchestrationInstancesHistory;
+        private readonly IDurableOrchestrationClient _durableOrchestrationClientMock;
+
+        public PurgeOrchestrationInstancesHistoryTests()
+        {
+            _purgeConfig = new PurgeOrchestrationInstancesHistoryConfiguration();
+            _purgeConfig.RuntimeStatuses = new OrchestrationRuntimeStatus[] { OrchestrationRuntimeStatus.Completed };
+            _timer = Substitute.For<TimerInfo>(default, default, default);
+            _logger = Substitute.For<ILogger>();
+            _definedNow = DateTime.UtcNow;
+            _purgeOrchestrationInstancesHistory = new PurgeOrchestrationInstancesHistory(
+                Options.Create(_purgeConfig),
+                () => _definedNow);
+            _durableOrchestrationClientMock = Substitute.For<IDurableOrchestrationClient>();
+        }
+
+        [Fact]
+        public async Task GivenValidInput_PurgeCompletedDurableFunctionsHistory_ShouldNotPurgeAnythingOrchestrationsAsync()
```

Comment:
nit: a more typical name in our repository would be using Given, When, Then:

Eg. `GivenNoInstances_WhenPurgingHistory_ThenSkipPurge`

Reply:
This comment goes for the rest: the "given" scenario is not valid input, but rather something more specific like no instances, no continuation, or multiple page